common/interpreter/wawaka_wasm/README.md

```diff
@@ -164,31 +164,92 @@ make benchmark
 
 Go to [contracts/wawaka/benchmarks](../../../contracts/wawaka/benchmarks/README.md), for more details.
 
-## Ahead-of-Time Compiled WASM
-** Disclaimer: Support for this feature is under active development and will be rolled-out at a later date. AoT mode cannot currently be enabled for testing. **
+## AoT-Compiled WASM
+
+Running ahead-of-time compiled WASM modules
+can significantly improve the performance of PDO contracts
+to near-native speeds. In internal benchmarks, we measured
+up to a 29x execution time speedup over interpreted WASM
+for some compute-intensive workloads when compiled ahead of
+time.
+
+### Code Deployment Integrity
+
+Since wawaka executes untrusted native code in AoT mode,
+PDO introduces new features to maintain both the confidentiality
+of contract data and the integrity of the contract enclave
+platform running the AoT-compiled contract. With our Code Deployment
+Integrity (CDI) framework, PDO requires WASM AoT compilers to
+produce a cryptographic proof binding the contract to the
+emitted AoT-compiled native code digitally signed by the compiler.
+
+PDO contract enclaves running wawaka in AoT mode specify a
+policy specifying trusted WASM AoT compilers, and ensure that
+any contract code they execute is accompanied by a CDI proof.
+Only if the contract enclave can validate the provenance and integrity
+of the AoT-compiled contract, does it execute the contract.
 
 ### Setup
 
+1. Set wawaka-aot mode:
+
 By default, wawaka will be built for interpreted wasm contracts.
-If you would like to enable
-ahead-of-time (AoT) compiled wasm contracts, set the environment variable
-`PDO_INTERPRETER`:
+To enable ahead-of-time (AoT) compiled WASM contracts, set the
+environment variable `PDO_INTERPRETER`:
 
 ```bash
 export PDO_INTERPRETER=wawaka-aot
 ```
 
-With AoT wasm enabled, the `wamrc` compiler (including dependencies)
+2. Set up the AoT compiler:
+
+Before running the CDI-aware AoT compilation, you will need to generate
+an ECDSA signing key:
+
+```bash
+openssl ecparam -name secp256k1 -genkey -noout -out <compiler signing key>.pem
+openssl ec -in <compiler signing key>.pem -pubout -out <compiler verifying key>.pem
+```
+
+With AoT WASM enabled, the `wamrc` compiler (including dependencies)
 shipped as part of the WAMR codebase will be built
 as part of the PDO build.
 
+We also provide `WawakaCDI`, a prototype tool for generating CDI proofs
+of contracts compiled with `wamrc`. `WawakaCDI` is built as part of the
+wawaka runtime during the PDO build.
+
+3. Specify the contract enclave CDI policy:
+
+The `pdo_policy.py` tool generates a JSON encoded contract enclave policy
+file, including a list of trusted AoT compiler signing keys.
+When in wawaka-aot mode, the eservice looks for this file and
+initializes the policy when the enclave is loaded.
+
+First, to generate a policy, run:
```

Comment:
this feels awfully complex... we are adding "yet another configuration file"

Reply:
I describe these steps here to be thorough, but I also automate this step (as noted in this doc later), so it's done automatically during `make install` for wawaka contracts.